compiler/src/dotty/tools/dotc/transform/localopt/StringContextChecker.scala

```diff
@@ -51,53 +58,52 @@ object StringContextMacro {
     def restoreReported() : Unit
   }
 
-  /** Interpolates the arguments to the formatting String given inside a StringContext
-   *
-   *  @param strCtxExpr the Expr that holds the StringContext which contains all the chunks of the formatting string
-   *  @param args the Expr that holds the sequence of arguments to interpolate to the String in the correct format
-   *  @return the Expr containing the formatted and interpolated String or an error/warning if the parameters are not correct
-   */
-  private def interpolate(strCtxExpr: Expr[StringContext], argsExpr: Expr[Seq[Any]])(using qctx: QuoteContext): Expr[String] = {
-    import qctx.tasty._
-    val sourceFile = strCtxExpr.unseal.pos.sourceFile
-
-    val (partsExpr, parts) = strCtxExpr match {
-      case Expr.StringContext(p1 as Consts(p2)) => (p1.toList, p2.toList)
-      case _ => report.throwError("Expected statically known String Context", strCtxExpr)
+  /** Check the format of the parts of the f".." arguments and returns the string parts of the StringContext */
+  def checkedParts(strContext_f: Tree, args0: Tree)(using Context): String = {
+
+    val (partsExpr, parts) = strContext_f match {
+      case TypeApply(Select(Apply(_, (parts: SeqLiteral) :: Nil), _), _) =>
+        (parts.elems, parts.elems.map { case Literal(Constant(str: String)) => str } )
+      case _ =>
+        report.error("Expected statically known String Context", strContext_f.srcPos)
+        return ""
     }
 
-    val args = argsExpr match {
-      case Varargs(args) => args
-      case _ => report.throwError("Expected statically known argument list", argsExpr)
+    val args = args0 match {
+      case args: SeqLiteral => args.elems
+      case _ =>
+        report.error("Expected statically known argument list", args0.srcPos)
+        return ""
     }
 
     val reporter = new Reporter{
```

Comment:
The name Reporter clashes with the existing Reporter trait in Dotty, would be better to use another name (and maybe also avoid the anonymous class here and use a named class)

Reply:
Done